server/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -414,6 +423,75 @@ private void createMetadataBasedIterables(Metadata metadata) {
         routines = () -> sequentialStream(routineInfos).filter(Objects::nonNull).iterator();
     }
 
+    public Set<String> enabledRoles(Role role, Roles roles) {
+        Set<String> enabledRoles = new HashSet<>();
+        enabledRoles.add(role.name());
+        for (GrantedRole grantedRole: role.grantedRoles()) {
+            enabledRoles.add(grantedRole.roleName());
+            Role retrievedRole = roles.findRole(grantedRole.roleName());
+            if (retrievedRole != null) {
+                if (!retrievedRole.grantedRoles().isEmpty()) {
+                    enabledRoles.addAll(enabledRoles(retrievedRole, roles));
+                }
+            }
+        }
+        return enabledRoles;
+    }
+
+    public Iterable<ApplicableRole> applicableRoles(Role role, Roles roles) {
+        List<ApplicableRole> applicableRoles = new ArrayList<>();
+
+        for (GrantedRole grantedRole: role.grantedRoles()) {
+            ApplicableRole newRole = new ApplicableRole();
+            newRole.setGrantee(role.name());
+            newRole.setRoleName(grantedRole.roleName());
+            newRole.setGrantable(roles.hasPrivilege(role, Permission.AL, Securable.CLUSTER, null));
+            applicableRoles.add(newRole);
+        }
+
+        return applicableRoles;
+    }
+
+    public Iterable<RoleTableGrant> roleTableGrants(Role role, Roles roles) {
+        List<RoleTableGrant> roleTableGrants = new ArrayList<>();
+        role.privileges().forEach(p -> {
+            RoleTableGrant roleTableGrant = new RoleTableGrant();
+            Securable securableType = p.subject().securable();
+            if (securableType.equals(Securable.TABLE)) {
+                roleTableGrant.setGrantor(p.grantor());
+
+                String[] fqn = Objects.requireNonNull(p.subject().ident()).split("\\.");
+                roleTableGrant.setTableSchema(fqn[0]);
+                roleTableGrant.setTableName(fqn[1]);
+                roleTableGrant.setGrantable(roles.hasPrivilege(role, Permission.AL, Securable.CLUSTER, null));
+                roleTableGrant.setGrantee(role.name());
+                roleTableGrant.setPrivilegeType(p.subject().permission().name());
+                roleTableGrant.setWithHierarchy(p.subject().permission().equals(Permission.DQL));
+                roleTableGrant.setTableCatalog("doc");
+
+                roleTableGrants.add(roleTableGrant);
+            }
+        });
+
+        return roleTableGrants;
+    }
+
+    public Iterable<AdministrableRoleAuthorization> administrableRoleAuthorizations(Role role, Roles roles) {
+        List<AdministrableRoleAuthorization> authorizations = new ArrayList<>();
+        for (GrantedRole grantedRole: role.grantedRoles()) {
```

Comment:
Maybe we need to use the `enabledRoles` here, so that we check all the granted roles in the hierarchy, have you checked how postgres behaves here?

Reply:
Postgres has very little documentation about this particular table, and very little information is available on the internet about this table. Hence I have been having a hard time reproducing this table and decided to go with roles with AL privileges instead, sticking to the Postgres doc as much as possible.

Any better suggestions or links are welcome.